### ISSUE TITLE
STDERR: dpkg: error: need an action option

### ISSUE THREAD

**Issue by gsaslis** [vkhatri/chef-filebeat#95]
Not really sure what has caused this breakage. 

here's the relevant portion of the log: 

    backend-builder: [2017-02-22T20:50:58+00:00] INFO: Processing directory[/etc/apt/preferences.d] action create (/tmp/packer-chef-client/cache/cookbooks/apt/providers/preference.rb line 42)
    backend-builder: [2017-02-22T20:50:58+00:00] INFO: Processing file[/etc/apt/preferences.d/filebeat.pref] action delete (/tmp/packer-chef-client/cache/cookbooks/apt/providers/preference.rb line 52)
    backend-builder: [2017-02-22T20:50:58+00:00] INFO: Processing file[/etc/apt/preferences.d/filebeat] action delete (/tmp/packer-chef-client/cache/cookbooks/apt/providers/preference.rb line 60)
    backend-builder: [2017-02-22T20:50:58+00:00] INFO: Processing file[/etc/apt/preferences.d/filebeat.pref] action create (/tmp/packer-chef-client/cache/cookbooks/apt/providers/preference.rb line 67)
    backend-builder: [2017-02-22T20:50:58+00:00] INFO: file[/etc/apt/preferences.d/filebeat.pref] created file /etc/apt/preferences.d/filebeat.pref
    backend-builder: [2017-02-22T20:50:58+00:00] INFO: file[/etc/apt/preferences.d/filebeat.pref] updated file contents /etc/apt/preferences.d/filebeat.pref
    backend-builder: [2017-02-22T20:50:58+00:00] INFO: file[/etc/apt/preferences.d/filebeat.pref] owner changed to 0
    backend-builder: [2017-02-22T20:50:58+00:00] INFO: file[/etc/apt/preferences.d/filebeat.pref] group changed to 0
    backend-builder: [2017-02-22T20:50:58+00:00] INFO: file[/etc/apt/preferences.d/filebeat.pref] mode changed to 644
    backend-builder: [2017-02-22T20:50:58+00:00] INFO: Processing apt_package[filebeat] action install (filebeat::install_package line 57)
    backend-builder:
    backend-builder: ================================================================================
    backend-builder: Error executing action `install` on resource 'apt_package[filebeat]'
    backend-builder: ================================================================================
    backend-builder:
    backend-builder: Mixlib::ShellOut::ShellCommandFailed
    backend-builder: ------------------------------------
    backend-builder: Expected process to exit with [0], but received '100'
    backend-builder: ---- Begin output of ["apt-get", "-q", "-y", "-o", "Dpkg::Options::='--force-confnew'", "install", "filebeat=1.3.1"] ----
    backend-builder: STDOUT: Reading package lists...
    backend-builder: Building dependency tree...
    backend-builder: Reading state information...
    backend-builder: The following NEW packages will be installed:
    backend-builder: filebeat
    backend-builder: 0 upgraded, 1 newly installed, 0 to remove and 135 not upgraded.
    backend-builder: Need to get 4255 kB of archives.
    backend-builder: After this operation, 12.6 MB of additional disk space will be used.
    backend-builder: Get:1 https://packages.elastic.co/beats/apt/ stable/main filebeat amd64 1.3.1 [4255 kB]
    backend-builder: Fetched 4255 kB in 1s (3414 kB/s)
    backend-builder: STDERR: dpkg: error: need an action option
    backend-builder:
    backend-builder: Type dpkg --help for help about installing and deinstalling packages [*];
    backend-builder: Use 'apt' or 'aptitude' for user-friendly package management;
    backend-builder: Type dpkg -Dhelp for a list of dpkg debug flag values;
    backend-builder: Type dpkg --force-help for a list of forcing options;
    backend-builder: Type dpkg-deb --help for help about manipulating *.deb files;
    backend-builder:
    backend-builder: Options marked [*] produce a lot of output - pipe it through 'less' or 'more' !
    backend-builder: E: Sub-process /usr/bin/dpkg returned an error code (2)
    backend-builder: ---- End output of ["apt-get", "-q", "-y", "-o", "Dpkg::Options::='--force-confnew'", "install", "filebeat=1.3.1"] ----
    backend-builder: Ran ["apt-get", "-q", "-y", "-o", "Dpkg::Options::='--force-confnew'", "install", "filebeat=1.3.1"] returned 100
    backend-builder:
    backend-builder: Resource Declaration:
    backend-builder: ---------------------
    backend-builder: # In /tmp/packer-chef-client/cache/cookbooks/filebeat/recipes/install_package.rb
    backend-builder:
    backend-builder: 57: package 'filebeat' do # ~FC009
    backend-builder: 58:   version version_string
    backend-builder: 59:   options node['filebeat']['apt']['options'] if node['filebeat']['apt']['options'] && node['platform_family'] == 'debian'
    backend-builder: 60:   notifies :restart, 'service[filebeat]' if node['filebeat']['notify_restart'] && !node['filebeat']['disable_service']
    backend-builder: 61:   flush_cache(:before => true) if node['platform_family'] == 'rhel'
    backend-builder: 62:   allow_downgrade true if node['platform_family'] == 'rhel'
    backend-builder: 63: end
    backend-builder:
    backend-builder: Compiled Resource:
    backend-builder: ------------------
    backend-builder: # Declared in /tmp/packer-chef-client/cache/cookbooks/filebeat/recipes/install_package.rb:57:in `from_file'
    backend-builder:
    backend-builder: apt_package("filebeat") do
    backend-builder: package_name "filebeat"
    backend-builder: action [:install]
    backend-builder: retries 0
    backend-builder: retry_delay 2
    backend-builder: default_guard_interpreter :default
    backend-builder: declared_type :package
    backend-builder: cookbook_name "filebeat"
    backend-builder: recipe_name "install_package"
    backend-builder: version "1.3.1"
    backend-builder: options "-o Dpkg::Options::='--force-confnew'"
    backend-builder: end
    backend-builder:
    backend-builder: Platform:
    backend-builder: ---------
    backend-builder: x86_64-linux
    backend-builder:
    backend-builder: [2017-02-22T20:51:00+00:00] INFO: Running queued delayed notifications before re-raising exception

This happens on both 0.3.9 and 0.4.8 (that i've tried). 

I am basically using the default (apt) attribute values (i.e. not overriding really anything through environment), and this is all on ubuntu 14.04. 

Any more specific info, just let me know plz

**Comment by gsaslis** [comment]
any ideas here?

**Comment by gsaslis** [comment]
ok worked it out. 

tl;dr 
it is due to an issue in chef-client version `12.19.33`, which is *solved* in `12.19.36`.

more info: https://github.com/chef/chef/issues/5836
originally found it through here: https://github.com/bdangit/chef-influxdb/issues/141
